backend_new/test/unit/services/listing.service.spec.ts

```diff
@@ -0,0 +1,1972 @@
+        { id: '2', name: 'listing 3' },
+        { id: '3', name: 'listing 4' },
+        { id: '4', name: 'listing 5' },
+        { id: '5', name: 'listing 6' },
+        { id: '6', name: 'listing 7' },
+        { id: '7', name: 'listing 8' },
+        { id: '8', name: 'listing 9' },
+        { id: '9', name: 'listing 10' },
+      ],
+      meta: {
+        currentPage: 1,
+        itemCount: 10,
+        itemsPerPage: 10,
+        totalItems: 10,
+        totalPages: 1,
+      },
+    });
+
+    expect(prisma.listings.findMany).toHaveBeenCalledWith({
+      skip: 0,
+      take: undefined,
+      orderBy: undefined,
+      where: {
+        AND: [],
+      },
+      include: {
+        jurisdictions: true,
+        listingsBuildingAddress: true,
+        reservedCommunityTypes: true,
+        listingImages: {
+          include: {
+            assets: true,
+          },
+        },
+        listingMultiselectQuestions: {
+          include: {
+            multiselectQuestions: true,
+          },
+        },
+        listingFeatures: true,
+        listingUtilities: true,
+        applicationMethods: {
+          include: {
+            paperApplications: {
+              include: {
+                assets: true,
+              },
+            },
+          },
+        },
+        listingsBuildingSelectionCriteriaFile: true,
+        listingEvents: {
+          include: {
+            assets: true,
+          },
+        },
+        listingsResult: true,
+        listingsLeasingAgentAddress: true,
+        listingsApplicationPickUpAddress: true,
+        listingsApplicationDropOffAddress: true,
+        units: {
+          include: {
+            unitAmiChartOverrides: true,
+            unitTypes: true,
+            unitRentTypes: true,
+            unitAccessibilityPriorityTypes: true,
+            amiChart: {
+              include: {
+                jurisdictions: true,
+                amiChartItem: true,
+                unitGroupAmiLevels: true,
+              },
+            },
+          },
+        },
+      },
+    });
+
+    expect(prisma.listings.count).toHaveBeenCalledWith({
+      where: {
+        AND: [],
+      },
+    });
+  });
+
+  it('testing list() with params', async () => {
+    prisma.listings.findMany = jest.fn().mockResolvedValue(mockListingSet(10));
+
+    prisma.listings.count = jest.fn().mockResolvedValue(20);
+
+    const params: ListingsQueryParams = {
+      view: 'base',
+      page: 2,
+      limit: 10,
+      orderBy: [ListingOrderByKeys.name],
+      orderDir: [OrderByEnum.ASC],
+      search: 'simple search',
+      filter: [
+        {
+          [ListingFilterKeys.name]: 'Listing,name',
+          $comparison: Compare.IN,
+        },
+        {
+          [ListingFilterKeys.bedrooms]: 2,
+          $comparison: Compare['>='],
+        },
+      ],
+    };
+
+    expect(await service.list(params)).toEqual({
+      items: [
+        { id: '0', name: 'listing 1' },
```

Comment:
can this be mocklistingset?

Reply:
covered: https://github.com/bloom-housing/bloom/pull/3458#discussion_r1212213566